### ISSUE TITLE
[Feature Request] Step by single page in dual page mode

### ISSUE THREAD

**Issue by woky** [johnfactotum/foliate#187]
`ebook-viewer` in Calibre had nice feature in dual page view where when you navigated to the next page, it actually really moved by single page only. Example:

_Next_ in Foliate (and almost every other viewer):
```
[A B] C D E F
A B [C D] E F
A B C D [E F]
```
_Next_ in `ebook-viewer` from Calibre:
```
[A B] C D E F
A [B C] D E F
A B [C D] E F
A B C [D E] F
A B C D [E F]
```
`ebook-viewer`'s approach is awesome because when you switch to next page you're left with context of what you just read. So this feature would be really great it Foliate. :-)

**Comment by johnfactotum** [comment]
I don't think this is currently supported by Epub.js, although I imagine that this won't be too hard to implement. You should [open an issue upstream](https://github.com/futurepress/epub.js/issues).

**Comment by johnfactotum** [comment]
It's actually very easy to do this in Epub.js. Just need to change one line:
```diff
diff --git a/src/assets/epub.js b/src/assets/epub.js
index 10e3370..a03096a 100644
--- a/src/assets/epub.js
+++ b/src/assets/epub.js
@@ -16350,7 +16350,7 @@ var Layout = function () {
 
                        spreadWidth = columnWidth * divisor + gap;
 
-                       delta = width;
+                       delta = width / 2;
 
                        this.width = width;
                        this.height = height;
```

It'll take more changes to make it optional, but for now you can try this patch and see if it fits your needs.

**Comment by digitalethics** [comment]
> it keeps you with context of what you've just read

I agree that the ability to not lose context while reading is very important. This can also be visually enhanced by gradually sliding a page out of the active window while a new one gradually slides into it.
